prow/plugins/invalidcommitmsg/invalidcommitmsg.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package invalidcommitmsg adds the "do-not-merge/invalid-commit-message"
+// label on PRs containing commit messages with @mentions or
+// keywords that can automatically close issues.
+package invalidcommitmsg
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+	"k8s.io/test-infra/prow/plugins/dco"
+)
+
+const (
+	pluginName            = "invalidcommitmsg"
+	invalidCommitMsgLabel = "do-not-merge/invalid-commit-message"
+	commentBody           = `[Keywords](https://help.github.com/articles/closing-issues-using-keywords) which can automatically close issues and at(@) mentions are not allowed in commit messages.
+
+**The list of commits with invalid commit messages**:
+
+%s
+
+<details>
+
+%s
+</details>
+`
+)
+
+var (
+	closeIssueRegex = regexp.MustCompile(`((?i)(clos(?:e[sd]?))|(fix(?:(es|ed)?))|(resolv(?:e[sd]?)))[\s:]+(\w+/\w+)?#(\d+)`)
+	atMentionRegex  = regexp.MustCompile(`\s([@][\w_-]+)`)
```

Comment:
`\b` is probably more appropriate than `\s` here.

Reply:
Did you mean `\B`? Updated it to use `\B`, PTAL.

Reply:
I meant `\b`, but it doesn't work the way I expected in this case because `@` does not match `\w`.  https://github.com/google/re2/wiki/Syntax

`\B` might be the best option then? I'm not sure if there is a practical difference between using `\B` or `(^|\s)` here.